File: MoviesExample/core/src/main/java/com/novoda/demo/movies/api/MoviesApi.java
package com.novoda.demo.movies.api;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;
import rx.Single;

public interface MoviesApi {

    String API_KEY = "ENTER_YOUR_API_KEY";

    @GET("movie/top_rated?api_key="+API_KEY)
    Call<MoviesResponse> topRated(@Query("page") int page);

    @GET("movie/{id}/videos?api_key="+API_KEY)
    Single<VideosResponse> videos(@Path("id") String movieId);
}

File: MoviesExample/core/src/main/java/com/novoda/demo/movies/api/MoviesApi.kt
package com.novoda.demo.movies.api

import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query
import rx.Single

const val API_KEY = "ENTER_YOUR_API_KEY"

interface MoviesApi {

    @GET("movie/top_rated?api_key=$API_KEY")
    fun topRated(@Query("page") page: Int): Call<MoviesResponse>

    @GET("movie/{id}/videos?api_key=$API_KEY")
    fun videos(@Path("id") movieId: String): Single<VideosResponse>

}
